### PR TITLE
Refactor MKL and oneDPL find package and add MKL_VERSION_2024 variable

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
This PR includes refactoring of `MKL` and `oneDPL` find_package logic and using `MKL_VERSION_2024 ` variable to target linking in `CMakeLists` files. 
It updates the logic to provide build compatibility with versions `2024.0.0` and `2023.2.0` for external and internal CI 

- [X] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

**Review by oleksandr-pavlyk — APPROVED** [review]
Thank you @vlad-perevezentsev . Nice solution

**Review by oleksandr-pavlyk — CHANGES_REQUESTED** [review]
Actually I spoke too soon. 

The `find_package(MKL QUIET)` would succeed even for MKL 2023.2. We need a different check to verify if it is MKL_2024. 

Can we check directly whether `MKL::MKL_SYCL::LAPACK` is defined?

**Comment by vlad-perevezentsev** [comment]
@oleksandr-pavlyk 
I overlooked the fact that MKL_FOUND is true on Linux for any MKL version  
I added an additional check `if(TARGET MKL::MKL_SYCL)` inside `IF(MKL_FOUND)` to identify the new version accurately

File: CMakeLists.txt [hunk 1]
```diff
@@ -36,7 +36,7 @@ set(MKL_THREADING "tbb_thread")
 set(MKL_VERSION_2024 FALSE)
 find_package(MKL QUIET)
 if(MKL_FOUND)
-    if(TARGET MKL::MKL_SYCL)
+    if(MKL_VERSION GREATER_EQUAL 2024.0.0)
```
Comment (oleksandr-pavlyk) [inline CMakeLists.txt:39]: Did you try

```
if (MKL_VERSION VERSION_GREATER_EQUAL "2024.0.0")
```

Comment (vlad-perevezentsev) [reply CMakeLists.txt:39]: @oleksandr-pavlyk
I just tested this and it works too.
Should I change checks?

Comment (oleksandr-pavlyk) [reply CMakeLists.txt:39]: Yes, please

Comment (vlad-perevezentsev) [reply CMakeLists.txt]: done

**Review by oleksandr-pavlyk — APPROVED** [review]
Thank you @vlad-perevezentsev

**Pull request merged**